### PR TITLE
Add a new board: Space Cubics OBC module 1

### PR DESCRIPTION
**Pull request by yashi**
This PR is to add our OBC as a Zephyr target board.

### CODE REVIEW

**Comment by yashi** [comment]
@henrikbrixandersen, this on-board computer is a single board based on Arm Cortex-M3 DesignStart FPGA-Xilinx edition.  Please let me know if you find anything bad.  Major difference from dev kits are
- We don't use XIP.  Zephyr image is copied from flash memory by FPGA
- We use HRMEM, High Reliable memory instead of TCM.  From software point of view, it just looks like SRAM

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Interesting! Without going into further details, I can see that the board documentation is missing, which makes it difficult to compare code against technical specifications.

Please add board documentation and re-request review.

**Comment by yashi** [comment]
@henrikbrixandersen, sorry for the delay.  I've added doc for the board.  Please review.  Thanks.

File: soc/arm/arm/designstart/Kconfig.soc
Comment (henrikbrixandersen) [reply soc/arm/arm/designstart/Kconfig.soc:17]: Yes, I think you should change both.

**Comment by yashi** [comment]
@henrikbrixandersen, do you have any concern?  I'd like to have any feedback.

File: boards/arm/scobc_module1/Kconfig.defconfig
Comment (henrikbrixandersen) [inline boards/arm/scobc_module1/Kconfig.defconfig:25]:
```suggestion
config UART_XLNX_UARTLITE
	depends on SERIAL
	default y
```

File: boards/arm/scobc_module1/scobc_module1_defconfig
Comment (henrikbrixandersen) [inline boards/arm/scobc_module1/scobc_module1_defconfig:18]: Are these needed?

**Review by henrikbrixandersen — APPROVED** [review]
Looks good (and very interesting), thank you. I have added a minor suggestion and a question, but nothing blocking.

File: boards/arm/scobc_module1/Kconfig.defconfig
Comment (yashi) [reply boards/arm/scobc_module1/Kconfig.defconfig:25]: done

File: boards/arm/scobc_module1/scobc_module1_defconfig
Comment (yashi) [reply boards/arm/scobc_module1/scobc_module1_defconfig:18]: Yes, these three are needed when `XIP=n`.

`CONFIG_FLASH_SIZE` is used in `zephyr/include/arch/arm/aarch32/cortex_m/scripts/linker.ld` and if you omit it, you get
```
In file included from /home/yashi/src/zephyr-project/zephyr/soc/arm/arm/designstart/linker.ld:9:
/home/yashi/src/zephyr-project/zephyr/include/arch/arm/aarch32/cortex_m/scripts/linker.ld:31:48: error: operator '==' has no left operand
   31 | #if !defined(CONFIG_XIP) && (CONFIG_FLASH_SIZE == 0)
      |                                                ^~
```

`CONFIG_FLASH_BASE_ADDRESS` and `CONFIG_BOOTLOADER_SRAM_SIZE` are needed to generate a proper `build/zephyr/misc/generated/configs.c`.  Because we have in `arch/Kconfig`

```
config FLASH_BASE_ADDRESS
	hex "Flash Base Address"
	default $(dt_chosen_reg_addr_hex,$(DT_CHOSEN_Z_FLASH)) if (XIP && (ARM || ARM64)) || !ARM
```
You won't get the value when `XIP=n`, but get
```
GEN_ABSOLUTE_SYM_KCONFIG(CONFIG_FLASH_BASE_ADDRESS, 0x);
```

With this wrong `config.c`, you get

```
/tmp/cczOtlxh.s: Assembler messages:
/tmp/cczOtlxh.s:127: Error: missing expression
```

This has been tested with
```shell
rm -rf build && west build -p -b scobc_module1 zephyr/samples/hello_world/
```

**Comment by yashi** [comment]
@henrikbrixandersen, Thank you for your review and Happy new year!

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by yashi** [comment]
So, what else do I need to get this one merged? Would someone enlighten me?

**Comment by henrikbrixandersen** [comment]
> So, what else do I need to get this one merged? Would someone enlighten me?

Now you wait for additional reviews and approvals. A PR requires at least two approvals before being merged into Zephyr.

**Review by nashif — APPROVED** [review]

**Pull request merged**
